### PR TITLE
Gidk 3dc1z6 show nav stage

### PR DESCRIPTION
**Pull request by shalinee-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Review by aditi-tiwari17 — APPROVED** [review]

File: apps/web-giddh/src/app/vat-report/vatReport.component.html [hunk 1]
```diff
@@ -1,49 +1,19 @@
...
-                <div class="row mr-0 ml-0">
-                    <div class="vat-report w-100" id="setting-tab">
-                        <tabset #staticTabs>
-                            <tab heading="{{commonLocaleData?.tab_heading}}">
-                                <div *ngIf="currentCompanyBranches?.length > 2 && currentOrganizationType === 'COMPANY'"
-                                    class="row mb-3 mt-3">
-                                    <div class="tax-heading col-md-3 col-sm-3 col-12">
-                                        <label>{{commonLocaleData?.entity}}:</label>
-                                    </div>
-                                    <div class="tax-dropdown col-md-9 col-sm-9 col-12 linked-entities">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vatReport.component.html:16]: This code getting removed. Need to revert

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/reports/components/reverse-charge-report-component/reverse-charge-report.component.ts [hunk 2]
```diff
@@ -209,6 +229,34 @@ export class ReverseChargeReport implements OnInit, OnDestroy {
...
+        }
+    }
+
+    /**
+      * This will toggle the GST popup
+      *
+      * @param {*} [event]
+      * @memberof ReverseChargeReport
+      */
+    public toggleGstPane(event?): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/reports/components/reverse-charge-report-component/reverse-charge-report.component.ts:252]:
```suggestion
    public toggleGstPane(event?: any): void {
```

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 3]
```diff
@@ -215,7 +215,12 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
     public commonLocaleData: any = {};
     /** This holds the active locale */
     public activeLocale: string = "";
-
+    /** True if sidebar is expanded */
+    public isSidebarExpanded: boolean = false;
+    /** This will hold if setting icon is disabled */
+    public isSettingsIconDisabled: boolean = false;
+    /* This will hold if resolution is more than 767 to consider as ipad screen */
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/header.component.ts:222]: Incorrect doc: `isIpadScreen` is set on 768px but in doc 767 is mentioned
`this.isIpadScreen = result?.breakpoints['(max-width: 768px)'];`

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 4]
```diff
@@ -789,10 +809,14 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
 
             if (this.asideSettingMenuState === "in") {
                 document.querySelector('body').classList.add('mobile-setting-sidebar');
-            } else {
+            }
+            else {
                 document.querySelector('body').classList.remove('mobile-setting-sidebar');
             }
-        }, (this.asideSettingMenuState === 'out') ? 100 : 0);
+
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/header.component.ts:816]: Avoid introducing empty lines

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 5]
```diff
@@ -1370,27 +1395,90 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
...
+            } else if (
+                document.getElementsByTagName("tabset") &&
+                document.getElementsByTagName("tabset").length > 0 &&
+                !this.router.url.includes("/vendor") && (!document.getElementsByClassName("static-tabs-on-page") || (document.getElementsByClassName("static-tabs-on-page") && document.getElementsByClassName("static-tabs-on-page").length === 0))) {
                 document.querySelector('body').classList.add('page-has-tabs');
                 document.querySelector('body').classList.remove('on-setting-page');
                 document.querySelector('body').classList.remove('on-user-page');
                 document.querySelector('body').classList.remove('mobile-setting-sidebar');
             }
+            /* this code is not working so that inventory sidebar is not working on mobile view, developer please check it */
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/header.component.ts:1413]: Remove this comment if it is done

File: apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    *
+    * @param {*} [event]
+    * @memberof TaxSidebarComponent
+    */
+    public closeAsidePaneIfMobile(event?): void {
+        if (this.isMobileScreen && event && event.target.className !== "icon-bar") {
+            this.closeAsideEvent.emit(event);
+        }
+    }
+    /**
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts:95]: Do leave blank line between two methods

File: apps/web-giddh/src/app/vat-report/transactions/vatReportTransactions.component.ts [hunk 7]
```diff
@@ -95,6 +111,8 @@ export class VatReportTransactionsComponent implements OnInit, OnDestroy {
     public ngOnDestroy(): void {
         this.destroyed$.next(true);
         this.destroyed$.complete();
+        document.querySelector('body').classList.remove('gst-sidebar-open');
+        this.asideGstSidebarMenuState === 'out'
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/vat-report/transactions/vatReportTransactions.component.ts:115]: Do add semicolon

File: apps/web-giddh/src/app/vat-report/vatReport.component.html [hunk 8]
```diff
@@ -1,49 +1,19 @@
 <section appTranslate [file]="'vat-report'" (localeData)="localeData = $event"
     (commonLocaleData)="commonLocaleData = $event">
-    <nav class="navbar">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/vat-report/vatReport.component.html:3]: Do check the `nav` opening tag is removed but can't find the closing `</nav>` tag getting removed in diff

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 9]
```diff
@@ -7693,30 +7692,31 @@ span#HW_badge_cont:nth-child(3) {
...
+    display: block;
 }
 .menu_open aside-setting.setting-sidebar {
-    left: 0 !important;
+    right: 0 !important;
 }
 .on-setting-page aside-setting.setting-sidebar,
 .on-user-page aside-setting.setting-sidebar {
-    left: -250px;
+   // left: -250px;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/assets/css/style-2.scss:7708]: Do remove the whole style, avoid empty CSS rules in scss files

**Review by prateek-giddh — CHANGES_REQUESTED** [review]
Please check all the comments and make them on all the environments

File: apps/web-giddh/src/app/vat-report/vatReport.component.html (hunk 1 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/vat-report/vatReport.component.html:16]: done

File: apps/web-giddh/src/app/shared/header/header.component.ts (hunk 3 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/shared/header/header.component.ts:222]: Done

File: apps/web-giddh/src/app/reports/components/reverse-charge-report-component/reverse-charge-report.component.ts (hunk 2 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/reports/components/reverse-charge-report-component/reverse-charge-report.component.ts:252]: This code is removed, no needed

File: apps/web-giddh/src/app/shared/header/header.component.ts (hunk 4 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/shared/header/header.component.ts:816]: Done

File: apps/web-giddh/src/app/shared/header/header.component.ts (hunk 5 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/shared/header/header.component.ts:1413]: Removed code, this is new inventory code it should be test env only.

File: apps/web-giddh/src/app/vat-report/transactions/vatReportTransactions.component.ts (hunk 7 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/vat-report/transactions/vatReportTransactions.component.ts:115]: Done

File: apps/web-giddh/src/app/vat-report/vatReport.component.html (hunk 8 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/vat-report/vatReport.component.html:3]: Added

File: apps/web-giddh/src/assets/css/style-2.scss (hunk 9 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/assets/css/style-2.scss:7708]: Removed

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 10]
```diff
@@ -1745,4 +1745,15 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
         text = text?.replace("[PLAN_NAME]", this.subscribedPlan?.planDetails?.name)?.replace("[PLAN_START_DATE]", this.subscribedPlan?.startedAt);
         return text;
     }
+    public showGstIcon(): boolean {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/header.component.ts:1743]: Consider using this and do add JS doc
```
public showGstIcon(): boolean {
    return this.currentPageUrl?.indexOf('pages/gstfiling') > -1 ||
        this.currentPageUrl?.indexOf('pages/reports/reverse-charge') > -1 ||
        this.currentPageUrl?.indexOf('pages/invoice/ewaybill') > -1;
}
```

**Review by prateek-giddh — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**